packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -1077,11 +1078,32 @@ export default function useAutocomplete(props) {
       tabIndex: -1,
       ...(!readOnly && { onDelete: handleTagDelete(index) }),
     }),
-    getListboxProps: () => ({
+    getListboxProps: ({ onScroll } = {}) => ({
       role: 'listbox',
       id: `${id}-listbox`,
       'aria-labelledby': `${id}-label`,
       ref: handleListboxRef,
+      onScroll: (event) => {
+        const listboxNode = listboxRef.current;
+        const listboxHeight = listboxNode.offsetHeight;
+        const contentHeight = listboxNode.scrollHeight;
+        const scrollPosition = listboxNode.scrollTop;
+
+        const listboxPaddingBottom = Number(
+          window.getComputedStyle(listboxNode).getPropertyValue('padding-bottom').replace('px', ''),
+        );
+
+        const distanceFromBottom =
+          contentHeight - (scrollPosition + listboxHeight + listboxPaddingBottom);
+        //   In general, it's best to use a small threshold value when checking for the scroll position, since the scrollTop, offsetHeight, and scrollHeight values are often not perfectly accurate. For example, some browsers might round these values to the nearest integer, which can cause small discrepancies.
+        if (distanceFromBottom <= 1 && listboxHeight !== contentHeight && onScrollToBottom) {
```

Comment:
I dove a bit unit this onscroll handler performance. https://github.com/mdn/content/issues/18843 seems to cover the topic. I didn't see anything specifically wrong, outside of: we should bail out a lot sooner if `onScrollToBottom` is not defined.

Reply:
done.